src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -160,6 +161,30 @@ public async Task AcknowledgeBatchProcessed(long subscriptionId, string name, st
                 DocumentsToResend = docsToResend
             };
 
+            var state = _serverStore.Engine.CurrentState;
+            if (command.ChangeVector == nameof(Constants.Documents.SubscriptionChangeVectorSpecialStates.DoNotChange) &&
+                (state == RachisState.Leader || state == RachisState.Follower))
+            {
+                // there are no changes for this subscription but we still want to check if we are the node that is responsible for this task.
+                // we can do that locally if we have a functional cluster (in a leader or follower state).
+
+                using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    var rawDatabaseRecord = _serverStore.Cluster.ReadRawDatabaseRecord(context, command.DatabaseName);
+                    if (rawDatabaseRecord == null)
+                        throw new DatabaseDoesNotExistException($"Cannot set command value of type {nameof(AcknowledgeSubscriptionBatchCommand)} for database {command.DatabaseName}, because it does not exist");
```

Comment:
when we can get that? in a middle of delete? what is the expectation here? should we just return or exception is what we want?

Reply:
After the database was deleted. We are also checking deletion in progress in `AssertSubscriptionState`.
We want to end the subscription task as soon as we detect that it isn't needed to run anymore.

Reply:
I will create a separate PR for this